test/evergreen/code_change_report/code_change_info.py

```diff
@@ -140,22 +295,59 @@ def create_report_info(change_list: dict, coverage_data: dict):
+                                                       file_path=new_file,
+                                                       line_number=line.new_lineno)
+                    branch_coverage = find_covered_branches(coverage_data=coverage_data,
+                                                            file_path=new_file,
+                                                            line_number=line.new_lineno)
+                    line_info['count'] = line_coverage
+                    line_info['branches'] = branch_coverage
+                    if line.old_lineno < 0:
+                        # Ihe line was changed, so update the counts for the overall change
+                        if line_coverage >= 0 and is_useful_line(line.content):
+                            change_num_lines += 1
+                            if line_coverage > 0:
+                                change_num_lines_covered += 1
+                        change_num_branches += len(branch_coverage)
+                        change_num_branched_covered += get_num_branches_covered(branch_coverage_info=branch_coverage)
+
+                    if preprocessed_complexity_data:
+                        function_info = get_function_info(file_path=new_file,
+                                                          line_number=line.new_lineno,
+                                                          preprocessed_complexity_data=preprocessed_complexity_data,
+                                                          preprocessed_prev_complexity_data=preprocessed_prev_complexity_data,
+                                                          coverage_data=coverage_data)
+                        if function_info:
+                            if new_file not in changed_function_info:
+                                changed_function_info[new_file] = dict()
+                            function_name = function_info['name']
+                            changed_function_info[new_file][function_name] = function_info
 
                 lines_info.append(line_info)
 
             change_info['lines'] = lines_info
             change_info_list.append(change_info)
 
-        report[new_file] = change_info_list
+            file_change_list[new_file] = change_info_list
+
+    report = dict()
+    report['summary_info'] = {'num_lines': change_num_lines,
+                              'num_lines_covered': change_num_lines_covered,
+                              'num_branches': change_num_branches,
+                              'num_branches_covered': change_num_branched_covered}
+    report['change_info_list'] = file_change_list
+    report['changed_functions'] = changed_function_info
 
     return report
 
 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-c', '--coverage', required=True, help='Path to the gcovr json code coverage data file')
+    parser.add_argument('-m', '--metrix_complexity_data', help='Path to the Metrix++ complexity data csv file')
```

Comment:
Do these arguments need to be optional? It'd be simpler to require them and not need to check if they're available

Reply:
That makes sense. During the development, I didn't always have those files available, and so I made them optional. But now the work is (nearly) complete, they should always be there.

Reply:
I've made those options required.